docs/wip-plugins.md

```diff
@@ -0,0 +1,169 @@
+
+### Chatterino API
+
+All Chatterino functions are exposed in a global table called `c2`. The following members are available:
+
+#### `log(level, args...)`
+
+Writes a message to the Chatterino log. The `level` argument should be a
+`LogLevel` member. All `args` should be convertible to a string with
+`tostring()`.
+
+Example:
+
+```lua
+c2.log(c2.LogLevel.Warning, "Hello, this should show up in the Chatterino log by default")
+
+c2.log(c2.LogLevel.Debug, "Hello world")
+-- Equivalent to doing qCDebug(chatterinoLua) << "[pluginDirectory:Plugin Name]" << "Hello, world"; from C++
+```
+
+#### `LogLevel` enum
+
+This table describes log levels available to Lua Plugins. The values behind the names may change, do not count on them. It has the following keys:
+
+- `Debug`
+- `Info`
+- `Warning`
+- `Critical`
+
+#### `register_command(name, handler)`
+
+Registers a new command called `name` which when executed will call `handler`.
+Returns `true` if everything went ok, `false` if there already exists another
+command with this name.
+
+Example:
+
+```lua
+function cmdWords(ctx)
+    -- ctx contains:
+    -- words - table of words supplied to the command including the trigger
+    -- channelName - name of the channel the command is being run in
+    c2.system_msg(ctx.channelName, "Words are: " .. table.concat(ctx.words, " "))
+end
+
+c2.register_command("/words", cmdWords)
+```
+
+Limitations/known issues:
+
+- commands registered in functions, not in the global scope might not show up in the settings UI,
+  rebuilding the window content caused by reloading another plugin will solve this
+- spaces in command names aren't handled very well (https://github.com/Chatterino/chatterino2/issues/1517)
+
+#### `send_msg(channel, text)`
+
+Sends a message to `channel` with the specified text. Also executes commands.
+
+Example:
+
+```lua
+function cmdShout(ctx)
+    table.remove(ctx.words, 1)
+    local output = table.concat(ctx.words, " ")
+    c2.send_msg(ctx.channelName, string.upper(output))
+end
+c2.register_command("/shout", cmdShout)
+```
+
+Limitations/Known issues:
+
+- it is possible to trigger your own Lua command with this causing a potentially infinite loop
+
+#### `system_msg(channel, text)`
+
+Creates a system message and adds it to the twitch channel specified by
+`channel`. Returns `true` if everything went ok, `false` otherwise. It will
+throw an error if the number of arguments received doesn't match what it
+expects.
+
+Example:
+
+```lua
+local ok = c2.system_msg("pajlada", "test")
+if (not ok)
+    -- channel not found
+end
+```
+
+### Changed globals
+
+#### `load(chunk [, chunkname [, mode [, env]]])`
+
+This function behaves really similarity to Lua's `load`, however it does not allow for bytecode to be executed.
+It achieves this by forcing all inputs to be encoded with `UTF-8`.
+
+See [official documentation](https://www.lua.org/manual/5.4/manual.html#pdf-load)
+
+#### `execfile(filename)`
```

Comment:
maybe rename to something like `import` or `include` or `require`

Reply:
I think I'll go for `import` or `include`, as `require` is a different lua function that we don't promise to be compatible with